File: 백준/Silver/11403. 경로 찾기/경로 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    static int N;
    static int[][] graph;
    static boolean[][] visited;
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());

        graph = new int[N][N];
        visited = new boolean[N][N];
        for(int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            for(int j = 0; j < N; j++) {
                graph[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        for(int i = 0; i < N; i++) {
            bfs(i);
        }

       StringBuilder sb = new StringBuilder();
       for(int i = 0; i < N; i++) {
           for(int j = 0; j < N; j++) {
               sb.append(visited[i][j] ? 1 : 0).append(" ");
           }
           sb.append("\n");
        }

        System.out.println(sb.toString());
    
    }

    static void bfs(int start) {
        Queue<Integer> q = new LinkedList<>();
        q.add(start);

        while(!q.isEmpty()) {
            int a = q.poll();

            for(int i = 0; i < N; i++) {
                if(!visited[start][i]) {
                    if(graph[a][i] == 1) {
                        q.add(i);
                        visited[start][i] = true;
                    }
                }
            }
        }
    }
}

File: 백준/Silver/11403. 경로 찾기/경로 찾기.py
from collections import deque
import sys

input = sys.stdin.readline

N = int(input())

graph = []
for _ in range(N):
    graph.append(list(map(int, input().split())))

def bfs(start):
    q = deque()
    q.append(start)
    
    while q:
        a = q.popleft()
        for i in range(N):
            if visited[start][i] == 0 and graph[a][i] == 1:
                q.append(i)
                visited[start][i] = 1
                
visited = [[0] * N for _ in range(N)]
for i in range(N):
    bfs(i)

for v in visited:
    print(' '.join(map(str, v)))